src/Xenial.Tasty.Cli/Program.cs

```diff
@@ -0,0 +1,236 @@
+﻿using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.VisualStudio.Threading;
+
+using StreamJsonRpc;
+
+using Xenial.Delicious.Protocols;
+
+using static SimpleExec.Command;
+using static Xenial.Delicious.Utils.PromiseHelper;
+
+namespace Xenial.Delicious.Cli
+{
+    class Program
+    {
+        static async Task<int> Main(string[] args)
+        {
+            var rootCommand = new RootCommand();
+
+            var interactiveCommand = new Command("interactive")
+            {
+                Handler = CommandHandler.Create<string, CancellationToken>(Interactive)
+            };
+            interactiveCommand.AddAlias("i");
+            var arg = new Option<string>("--project");
+            arg.AddAlias("-p");
+            interactiveCommand.Add(arg);
+            rootCommand.Add(interactiveCommand);
+
+            return await rootCommand.InvokeAsync(args);
+        }
+
+        static async Task<int> Interactive(string project, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var path = Path.GetFullPath(project);
+                if (!string.IsNullOrEmpty(path) && Directory.Exists(path))
+                {
+                    var directoryName = new DirectoryInfo(path).Name;
+                    var csProjFileName = Path.Combine(path, $"{directoryName}.csproj");
+                    if (File.Exists(csProjFileName))
+                    {
+                        Console.WriteLine(csProjFileName);
+
+                        var connectionId = $"TASTY_{Guid.NewGuid()}";
+
+                        using var stream = new NamedPipeServerStream(connectionId, PipeDirection.InOut, NamedPipeServerStream.MaxAllowedServerInstances, PipeTransmissionMode.Byte, PipeOptions.Asynchronous);
```

Comment:
Be careful here, only one client connection per server, if you need an additional client to attach you need to spawn a new one.
You can share the named pipe tough

Reply:
As far as I know there is no way to bind one stream to more than one json RPC. So those should be coupled like in tasty core, I think.

Reply:
One thing I am really thinking about is support for named pipes on linux, how are named pipe streams are working without names? (cause you need to pass null as a name, otherwise it will throw a `NSE`)

Reply:
I think this is fine for now. We need a better pattern once we introduce websocket support